### PR TITLE
chore: @typescript-eslint/parser

### PR DESCRIPTION
**Pull request by mightyiam**

### CODE REVIEW

**Comment by mightyiam** [comment]
Could anyone please provide me with a hint on the failing `ExportMap.get`?

File: package.json [hunk 1]
```diff
@@ -68,8 +69,7 @@
     "redux": "^3.7.2",
     "rimraf": "^2.6.3",
     "sinon": "^2.4.1",
-    "typescript": "^3.2.2",
-    "typescript-eslint-parser": "^21.0.2"
+    "typescript": "^3.2.2"
```
Comment (mightyiam) [inline package.json]: I would expect this version to be pinned. TypeScript don't follow SemVer.

File: README.md [hunk 2]
```diff
@@ -148,7 +148,7 @@ rules:
 
 You may use the following shortcut or assemble your own config using the granular settings described below.
 
-Make sure you have installed the [`@typescript-eslint/parser`] which is used in the following configuration. Unfortunately NPM does not allow to list optional peer dependencies. 
+Make sure you have installed the [`@typescript-eslint/parser`] which is used in the following configuration. Unfortunately NPM does not allow to list optional peer dependencies.
```
Comment (ljharb) [inline README.md]:
```suggestion
Make sure you have installed the [`@typescript-eslint/parser`] which is used in the following configuration. Unfortunately npm does not support optional peer dependencies.
```

File: package.json (hunk 1 above)
Comment (ljharb) [reply package.json]: sure, fair enough:
```suggestion
    "typescript": "~3.2.2"
```

**Comment by mightyiam** [comment]
Would you mind if we support TypeScript only on ESLint =< 4 and Node.js =< 8 ?

**Comment by schmod** [comment]
FYI, I've added similar changes in:
https://github.com/benmosher/eslint-plugin-import/pull/1304

**Comment by ljharb** [comment]
@mightyiam eslint 4+, that sounds fine; i'm not sure why it wouldn't work on node 4 and 6?

**Comment by benmosher** [comment]
closing to defer to #1304, thanks for this though!

**Pull request closed without merging**